app/views/schedule/scheduled_publishing_datetime.html.erb

```diff
@@ -1,49 +1,53 @@
 <% content_for :back_link, render_back_link(href: document_path(@edition.document)) %>
 <% content_for :title, t("schedule.scheduled_publishing_datetime.title") %>
-
-<% current_scheduled_publish_datetime = @edition.revision.scheduled_publishing_datetime %>
-<% default_date = Time.zone.now.tomorrow %>
-<% publish_date = current_scheduled_publish_datetime || default_date %>
-<% schedule_param = params["schedule"] ? params["schedule"] : {} %>
-<% selected_time = schedule_param["time"] ||  current_scheduled_publish_datetime&.strftime("%l:%M%P")&.strip %>
-<% action_not_selected = @issues&.select { |i| i.issue_key == :action_not_selected }&.first&.to_item %>
-
+<%
+  current_scheduled_publish_datetime = @edition.scheduled_publishing_datetime
+  default_time = Time.current.tomorrow
+  publish_time = current_scheduled_publish_datetime || default_time
+  action_not_selected = @issues&.select { |i| i.issue_key == :action_not_selected }&.first&.to_item
+%>
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
-    <%= form_tag(save_scheduled_publishing_datetime_path(@edition.document),
-                 id: "scheduled_publishing_datetime") do %>
-      <% date_input_id = "input-#{SecureRandom.hex(4)}" %>
+    <%= form_tag save_scheduled_publishing_datetime_path(@edition.document) do %>
       <%= render "govuk_publishing_components/components/label", {
         bold: true,
         text: t("schedule.scheduled_publishing_datetime.date"),
-        html_for: date_input_id,
+        html_for: "scheduled-date",
       } %>
+      <%
+        errors = (@issues&.items_for(:scheduled_date) || [])
+        error_message = errors.map { |item| item[:text] }.join("<br/>").html_safe
+      %>
       <%= render "govuk_publishing_components/components/date_input", {
-        name: "schedule",
+        name: "schedule[scheduled_date]",
         hint: t("schedule.scheduled_publishing_datetime.date_hint_text"),
-        id: date_input_id,
+        id: "scheduled-date",
+        error_message: error_message.presence,
         items: [
           {
             name: "day",
             width: 2,
-            value: schedule_param["day"] || publish_date&.day
+            value: params.dig(:schedule, :scheduled_date, :day) || publish_time&.day
```

Comment:
I wonder if `publish_time` should be renamed `publish_datetime`?  When I first saw the variable I assumed it was the time only, and got confused when I saw it appear as a date value.

Reply:
Sure, I'll have a look at that - it's a bit of a pain as a Time object in Ruby always represents a date so that kind of makes sense to name it a time. But I know we also deal with time as a string.